libafl/src/feedbacks/map.rs

```diff
@@ -484,8 +486,33 @@ where
                 .filter(|(_, value)| *value != initial)
             {
                 history_map[i] = R::reduce(history_map[i], value);
+                filled += 1;
             }
         }
+
+        // at this point you are executing this code, the testcase is always interesting
+        let len = history_map.len();
+        // opt: if not tracking optimisations, we technically don't show the *current* history
+        // map but the *last* history map; this is better than walking over and allocating
+        // unnecessarily
+        manager.fire(
+            state,
+            Event::UpdateUserStats {
+                name: self.stats_name.to_string(),
+                value: UserStats::new(
+                    UserStatsValue::Ratio(
+                        self.novelties
+                            .as_ref()
+                            .map_or(filled, |novelties| filled + novelties.len())
```

Comment:
I think that `filled` here is only going to contain the covered indexes for this new testcase, rather than all the ones in the whole corpus (because you are calculating `filled` by iterating over the `observer` looking for set bytes, rather than the `history_map`).